src/rpc/rawtransaction.cpp

```diff
@@ -858,132 +965,180 @@ UniValue consolidatemsunspent(const UniValue& params, bool fHelp)
 
-    if (nMaxInputs < 1)
-        nMaxInputs = 1;
-
-    else if (nMaxValue < 0)
+    if (nMaxValue < 0)
         throw JSONRPCError(RPC_INVALID_PARAMETER, "Value must not be less then 0");
 
-    int nReqSigs = 0;
-    int64_t nRedeemScriptSize = 0;
+    CBitcoinAddress Address(sAddress);
 
-    if (nReqSigsType < 1 || nReqSigsType > 4)
-        throw JSONRPCError(RPC_INVALID_PARAMETER, "Invalid type of multi-signature address chosen");
+    // Check if the address is valid
+    if (!Address.IsValid())
+        throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "Invalid Gridcoin Address");
 
-    else if (nReqSigsType == 1)
     {
-        nReqSigs = 2;
-        nRedeemScriptSize = 210;
+        // This new method we should lock wallet as well with cs_main to make sure we don't deadlock
+        LOCK2(cs_main, pwalletMain->cs_wallet);
 
-        if (nMaxInputs > 40)
-            nMaxInputs = 40;
-    }
+        // Check The OP codes to determine how many signatures are needed.
+        // redeemScript is formatted as such <OP_CODE>21<pubkey>21<pubkey>..........<OP_CODE><OP_CHECKMULTISIG>
+        // We doing that here with this code since it's already in the wallet and verified. This saves cycles.
+        bool fOPCodeSuccess = false;
+        CScript CSDestSubscript;
 
-    else if (nReqSigsType == 2)
-    {
-        nReqSigs = 3;
-        nRedeemScriptSize = 278;
+        // Gather the Redeemscript for calculations
+        CScript CSDest;
+        txnouttype WhichType;
+        vector<valtype> vSolutions;
 
-        if (nMaxInputs > 30)
-            nMaxInputs = 30;
-    }
+        CSDest.SetDestination(Address.Get());
 
-    else if (nReqSigsType == 3)
-    {
-        nReqSigs = 3;
-        nRedeemScriptSize = 346;
+        // Solve the solutions for Destination script
+        if (!Solver(CSDest, WhichType, vSolutions))
+            throw JSONRPCError(RPC_WALLET_ERROR, "Solver failed to get solutions for CScript() of Gridcoin Address");
 
-        if (nMaxInputs > 30)
-            nMaxInputs = 30;
-    }
+        // Get the CScript subscript aka redeemScript from wallet which requires CScriptID supplied by vSolutions[0]
+        if (!pwalletMain->GetCScript(CScriptID(uint160(vSolutions[0])), CSDestSubscript))
+            throw JSONRPCError(RPC_WALLET_ERROR, "Failed to retrieve redeemScript from wallet.");
 
-    else if (nReqSigsType == 4)
-    {
-        nReqSigs = 4;
-        nRedeemScriptSize = 346;
+        // Set the redeemScript size in bytes.
+        nRedeemScriptSize = CSDestSubscript.size();
 
-        if (nMaxInputs > 20)
-            nMaxInputs = 20;
-    }
+        try
+        {
+            CScript::const_iterator script = CSDestSubscript.begin();
+            CScript::const_iterator scriptend = CSDestSubscript.end();
+            opcodetype Whatopcode;
+            valtype vValue;
 
-    CBitcoinAddress Address(sAddress);
+            while (script < scriptend)
```

Comment:
Style thing: If begin() and end() are supported, can you use a newer style range for loop?

Reply:
Nope. I am wrong. it is good as is.